tests/test_network_consistency.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import unittest
+from glob import glob
+from unittest.case import skipIf
+
+import torch
+from parameterized.parameterized import parameterized
+
+from monai.networks.nets.net_factory import NetworkFactory
+
+extra_test_data_dir = os.environ.get("MONAI_EXTRA_TEST_DATA", None)
+
+TESTS = []
+if extra_test_data_dir is not None:
+    for data_path in glob(os.path.join(extra_test_data_dir, "**", "*.pt")):
+        json_path = data_path[:-3] + ".json"
+        # net_name is filename until first underscore (e.g., unet_0.pt is unet)
+        net_name = os.path.basename(data_path).split("_")[0]
+        TESTS.append((net_name, data_path, json_path))
+
+
+class TestNetworkConsistency(unittest.TestCase):
+    @skipIf(
+        len(TESTS) == 0,
+        "To run these tests, clone https://github.com/Project-MONAI/MONAI-extra-test-data and set MONAI_EXTRA_TEST_DATA",
+    )
+    @parameterized.expand(TESTS, skip_on_empty=True)
+    def test_network_consistency(self, net_name, data_path, json_path):
+
+        print("Net name: " + net_name)
+        print("Data path: " + data_path)
+        print("JSON path: " + json_path)
+
+        # Load data
+        loaded_data = torch.load(data_path)
+
+        # Load json from file
+        json_file = open(json_path)
+        model_params = json.load(json_file)
+        json_file.close()
+
+        # Create model
+        factory = NetworkFactory()
```

Comment:
could we use these simple commands for now:
```py
import monai.networks.nets as nets
nets.__dict__['UNet']
```
I feel the network factory idea should have a separate discussion

Reply:
good suggestion, thanks!